src/lib/keypom.ts

```diff
@@ -201,18 +194,39 @@ class KeypomJS {
 
   getDropSupplyForOwner = async ({ accountId }) => await getDropSupplyForOwner({ accountId });
 
-  getDropMetadata = (metadata: string) =>
+  getDropMetadata = (metadata: string | undefined) =>
     JSON.parse(metadata || JSON.stringify({ dropName: 'Untitled' }));
 
   deleteDrops = async ({ wallet, dropIds }) => await deleteDrops({ wallet, dropIds });
 
   deleteKeys = async ({ wallet, dropId, publicKeys }) =>
     await deleteKeys({ wallet, dropId, publicKeys });
 
-  getDropInfo = async (dropId?: string): Promise<ProtocolReturnedDrop> => {
-    if (!dropId) throw new Error('Missing dropId');
-    // TODO: add basic catch
-    return await getDropInformation({ dropId });
+  getDropInfo = async ({
+    dropId,
+    secretKey,
+  }: {
+    dropId?: string;
+    secretKey?: string;
+  }): Promise<ProtocolReturnedDrop> => {
+    let drop: ProtocolReturnedDrop;
+
+    try {
+      if (dropId) {
+        drop = await getDropInformation({ dropId });
```

Comment:
i think we can add both parameters inside `getDropInformation({ dropId, secretKey })`